onnx/reference/ops/op_dequantize_linear.py

```diff
@@ -48,26 +48,41 @@ def reshape_input(
             raise ValueError("axis cannot be None.")
         if len(value.shape) == 0:
             return value
-        dims = [1] * len(shape)
-        try:
-            dims[axis] = value.size
-        except IndexError as e:
-            raise IndexError(
-                f"axis is out of boundary, axis={axis}, "
-                f"value.shape={value.shape}, shape={shape}."
-            ) from e
-        return value.reshape(tuple(dims))
+        if len(value.shape) == 1:
+            dims = [1] * len(shape)
+            try:
+                dims[axis] = value.size
+                return value.reshape(tuple(dims))
+            except IndexError as e:
+                raise IndexError(
+                    f"axis is out of boundary, axis={axis}, "
+                    f"value.shape={value.shape}, shape={shape}."
+                ) from e
+        if len(value.shape) >= 2:
+            if len(shape) != len(value.shape):
+                raise ValueError(
+                    "Input 2 must be a number, a vector or a tensor with the same rank as the input."
+                )
+            block_shape = np.array(shape) // np.array(value.shape)
+            if sum(block_shape != 1) != 1:
+                raise ValueError("Blocked quantization is defined for 1-D blocks only.")
+            block_dim = np.where(block_shape != 1)[0][0]
+            if shape[block_dim] % value.shape[block_dim] != 0:
+                raise ValueError(
+                    "Blocked quantization requires the scale dimensions to divide the input dimensions"
+                )
+            block_size = shape[block_dim] // value.shape[block_dim]
+            return np.repeat(value, repeats=block_size, axis=block_dim)
+
+        return None
 
-    def _run(
+    def common_run(
```

Comment:
should this be private?

Reply:
The entire class is defined as private, but sure, this method _could_ be private, so I guess it should be... :) 
Modified here and in `op_quantize_linear.py` for consistency.